src/foam/u2/view/TreeView.js

```diff
@@ -200,37 +207,44 @@ foam.CLASS({
         }).
         start().
           addClass(self.myClass('heading')).
-            style({
-              'padding-left': ((( self.level - 1) * 16 + 28) + 'px')
-            }).
-            add(this.slot( function(level, selected, id) {
-              if ( level === 1 ) {
-                var isDefault = ! this.data.icon || ! this.data.activeIcon;
-                var imgUrl = isDefault ? 'images/settings-icon-resting.svg' : self.data.icon;
-                if ( selected && foam.util.equals(selected.id, id) ) {
-                  imgUrl = isDefault ? 'images/settings-icon-active.svg' : self.data.activeIcon;
-                }
+          style({
+            'padding-left': ((( self.level - 1) * 16 + 28) + 'px')
+          }).
+          add(this.slot( function(level, selected, id) {
+            if ( level === 1 ) {
+              var imgUrl = self.data.icon;
+              if ( selected && foam.util.equals(selected.id, id) ) {
+                imgUrl = self.data.activeIcon;
+              }
+              if ( imgUrl )
                 return this.E().start('img').
                   addClass(self.myClass('label-icon')).
                   attrs({ 'src': imgUrl, 'width': '16px', 'height': '16px' }).
                 end();
-              }
-            }, self.level$, this.selection$, this.data$.dot('id'))).
-            start().
-              addClass(self.myClass('select-level')).
-              style({
-                'width': '100%',
-                'padding-right': '20px'
-              }).
-              addClass(self.myClass('label')).
+              else
+                return;
+            }
+          }, self.level$, this.selection$, this.data$.dot('id'))).
+          start().
+            addClass(self.myClass('select-level')).
+            style({
+              'width':         '100%',
+              'padding-right': '20px'
+            }).
+            start()
+              .addClass(self.myClass('label')).
               call(this.formatter, [self.data]).
-              start('span').
-              addClass('toggle-icon').
+            end().
+            start('span').
+              addClass(self.myClass('toggle-icon')).
               show(this.hasChildren$).
               style({
-                'visibility': 'visible',
-                'font-size':  '16px',
-                'transform':  this.expanded$.map(function(c) { return c ? 'rotate(180deg)' : 'rotate(90deg)'; })
+                'visibility':    'visible',
+                'font-size':     '16px',
+                'float':         'right',
+                'height':        '6px',
+                'padding':       self.expanded$.map(function(c) { return c ? '5px 0px 8px 3px' : '0px 0px 11px 4px'}),
```

Comment:
Again, I don't have a better solution

Reply:
That's fine. CSS isn't an elegant tool.

Reply:
What if we make toggle icon's width equal to its height? And if it's still off we can just add some padding to the bottom?

Reply:
thanks!